src/lib/memory/tracking_memory_resource.cpp

```diff
@@ -0,0 +1,26 @@
+#include "tracking_memory_resource.hpp"
+
+namespace opossum {
+
+void* TrackingMemoryResource::do_allocate(std::size_t bytes, std::size_t alignment) {
+  auto timestamp = std::chrono::system_clock::now();
+  _memory_timeseries.emplace_back(std::make_pair(timestamp, bytes));
+  return std::malloc(bytes); // NOLINT
```

Comment:
I think you should also use the alignment in this method. Some data structures might want to make better use of cache lines etc. Check: https://en.cppreference.com/w/cpp/memory/c/aligned_alloc

Reply:
Good idea!
What would be a suitable alignment size?

Reply:
I think you don't have to decide for one here, but you should take the `std::size_t alignment` from the parameters